.github/workflows/schedule.py

```diff
@@ -171,55 +172,47 @@ def create_issue(self, title_date: date) -> None:
-            else:
-                log.info('Running %r …', command)
-                process = subprocess.run(command, check=True, stdout=subprocess.PIPE)
-                url = process.stdout.decode().strip().splitlines()[-1]
-            # If the current token lacks the necessary permissions, the GitHub
-            # CLI silently fails to apply the requested labels or assignees. We
-            # therefore need to verify those.
-            log.info('… created %r, verifying labels and assignees …', url)
-            path = urllib.parse.urlparse(url).path.removeprefix('/')
-            actual_repository, dir, issue_number = path.rsplit('/', maxsplit=2)
-            assert dir == 'issues'
-            if repository is not None:
-                assert repository == actual_repository, (repository, actual_repository)
-            command = [
-                'gh', 'issue', 'view',
-                *flags,
-                '--json=labels,assignees',
-                issue_number
-            ]
-            if self.dry_run:
-                log.info('Would run %r', command)
-                issue = {
-                    'assignees': [{'login': assignee} for assignee in assignees],
-                    'labels': [{'name': label} for label in labels]
-                }
             else:
                 log.info('Running %r …', command)
                 process = subprocess.run(command, check=True, stdout=subprocess.PIPE)
                 issue = json.loads(process.stdout)
-            actual_assignees = set(map(itemgetter('login'), issue['assignees']))
-            assert set(assignees) == actual_assignees, (assignees, actual_assignees)
-            actual_labels = set(map(itemgetter('name'), issue['labels']))
-            assert set(labels) <= actual_labels, (labels, actual_labels)
-            log.info('Successfully created and verfied issue #%s', issue_number)
+                # If the current token lacks the necessary permissions, the GitHub
+                # CLI silently fails to apply the requested labels or assignees. We
+                # therefore need to verify those.
+                url = issue['html_url']
+                log.info('… created %r, verifying labels and assignees …', url)
+                path = urllib.parse.urlparse(url).path.removeprefix('/')
+                actual_repository, dir, issue_number = path.rsplit('/', maxsplit=2)
+                assert dir == 'issues'
+                assert repository == actual_repository, (repository, actual_repository)
+                actual_assignees = set(map(itemgetter('login'), issue['assignees']))
+                assert set(assignees) == actual_assignees, (assignees, actual_assignees)
+                actual_labels = set(map(itemgetter('name'), issue['labels']))
+                assert set(labels) <= actual_labels, (labels, actual_labels)
+                actual_type = map(itemgetter('name'), issue['type'])
+                assert type == actual_type, (type, actual_type)
```

Comment:
This looks wrong. How can `type` compare equal to a `map` generator instance? And does the `type` field really have a nested `name` field?

Reply:
This is an instance of `IssueType` from GitHub, with nested fields, see `name` at the bottom of the 'Fields for IssueType' heading in https://docs.github.com/en/graphql/reference/objects#issuetype.

Unfortunately, the sample response object for this [`gh api` create request](https://docs.github.com/en/rest/issues/issues?apiVersion=2022-11-28#create-an-issue) doesn't include the IssueType. But it is present when 1) enabled at the organization level, and 2) the issue has a type assigned.

Reply:
That still doesn't explain or justify the use of `map`.

On line 214, you assert that the local variables `type` and `actual_type` are equal. `type` is assigned on line 176 from the field of the same name of `self.properties`, which is annotated as having type `str`. Therefore, I presume that both `type` and `actual_type` are meant to be strings. However, after the assignment on line 213, `actual_type` will be an instance of `map`, not `str`.

Furthermore, the link provided doesn't show any field named "type" under the the `Issue` schema. I think the correct field name is `issueType`.

All things considered, I think the correct code would be

```suggestion
                actual_type = issue['issueType']['name']
                assert type == actual_type, (type, actual_type)
```

Reply:
You're correct that the use of `map` wasn't justified. Here's a [sample response](https://gitlab.dev.singlecell.gi.ucsc.edu/ucsc/azul/-/snippets/40/raw/main/gh-api-create-response.json) (GitLab dev)containing the type object.